AppSearchSample/app/src/main/java/com/android/example/appsearchsample/NoteListItemAdapter.kt

```diff
@@ -43,31 +52,44 @@ class NoteListItemAdapter(private val onDelete: (Note?) -> Unit) :
   }
 
   override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
-    holder.bind(getItem(position), onDelete)
+    holder.bind(getItem(position), onDelete, position)
   }
 
   /** ViewHolder for [NoteListItemAdapter]. */
-  class NoteViewHolder(
+  inner class NoteViewHolder(
     binding: ItemNoteBinding,
   ) : RecyclerView.ViewHolder(binding.root) {
     private val noteTextView = binding.noteText
     private val noteDeleteButtonView = binding.noteDeleteButton
 
-    fun bind(note: Note?, onDelete: (Note?) -> Unit) {
-      noteTextView.text = note?.text
-      noteDeleteButtonView.setOnClickListener { onDelete(note) }
+    fun bind(sr: SearchResult, onDelete: (SearchResult?) -> Unit, position: Int) {
+      val note = sr.genericDocument.toDocumentClass(Note::class.java)
+      val sb = SpannableStringBuilder(note.text)
+
+      sr.matchInfos.forEach {
+        sb.setSpan(StyleSpan(BOLD), it.exactMatchRange.start, it.exactMatchRange.end, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
+      }
+
+      noteTextView.text = sb
+
+      noteTextView.setBackgroundColor(Color.parseColor(note.color))
+
+      noteDeleteButtonView.setOnClickListener { onDelete(sr) }
     }
   }
 
   companion object {
-    private val NOTES_COMPARATOR = object : DiffUtil.ItemCallback<Note>() {
-      override fun areItemsTheSame(oldItem: Note, newItem: Note): Boolean {
-        return oldItem.id === newItem.id &&
-          oldItem.namespace === newItem.namespace
+    private val NOTES_COMPARATOR = object : DiffUtil.ItemCallback<SearchResult>() {
+      override fun areItemsTheSame(oldItem: SearchResult, newItem: SearchResult): Boolean {
+        val oi = oldItem.genericDocument.toDocumentClass(Note::class.java)
+        val ni = newItem.genericDocument.toDocumentClass(Note::class.java)
+        return oi.id === ni.id &&
+          oi.namespace === ni.namespace
       }
 
-      override fun areContentsTheSame(oldItem: Note, newItem: Note): Boolean {
-        return oldItem == newItem
+      override fun areContentsTheSame(oldItem: SearchResult, newItem: SearchResult): Boolean {
```

Comment:
Now that we're comparing SearchResult objects instead of Note objects, should our content equality check be updated to compare SearchResult objects?